nautobot/dcim/models/power.py

```diff
@@ -137,31 +174,119 @@ class PowerFeed(PrimaryModel, PathEndpoint, CableTermination):
 
+        # Destination panel validation
+        if self.destination_panel:
+            # Cannot feed into the same panel
+            if self.destination_panel == self.power_panel:
+                raise ValidationError({"destination_panel": "A power feed cannot connect a panel to itself"})
+            # TODO: add loop detection when graph structure is implemented for path tracing
+
+        # Enforce mutual exclusivity between cable connections and destination_panel
+        if self.destination_panel and self._path and self._path.destination:  # pylint: disable=no-member
+            raise ValidationError(
+                {
+                    "destination_panel": "Cannot specify a destination panel when the power feed is connected via cable. "
+                    "Power feeds can either connect to a panel OR be cabled to an endpoint, but not both."
+                }
+            )
+
+        if self._path and self._path.destination and self.destination_panel:  # pylint: disable=no-member
+            raise ValidationError(
+                {
+                    "__all__": "This power feed cannot be connected via cable when a destination panel is specified. "
+                    "Power feeds can either connect to a panel OR be cabled to an endpoint, but not both."
+                }
+            )
+
+        # Breaker position and pole validation
+        if self.breaker_position is not None:
+            # Default to single pole breaker when breaker_position is specified but breaker_pole_count is not
+            if self.breaker_pole_count is None:
+                self.breaker_pole_count = PowerFeedBreakerPoleChoices.POLE_1
+
+            # Get occupied positions once for both validations
+            occupied_positions = self.get_occupied_positions()
+
+            # Ensure breaker positions fit within panel capacity
+            if self.power_panel.breaker_position_count is not None:
+                if occupied_positions:
+                    max_occupied_position = max(occupied_positions)
+                    if max_occupied_position > self.power_panel.breaker_position_count:
+                        raise ValidationError(
+                            {
+                                "breaker_position": f"Breaker configuration starting at position {self.breaker_position} "
+                                f"with {self.breaker_pole_count} poles would occupy positions {sorted(occupied_positions)}, "
+                                f"but panel only has {self.power_panel.breaker_position_count} breaker positions"
+                            }
+                        )
+
+            # Check for breaker position conflicts with other feeds
+            conflicts = PowerFeed.objects.filter(
+                power_panel=self.power_panel, breaker_position__isnull=False, breaker_pole_count__isnull=False
+            ).exclude(pk=self.pk if self.pk else None)
```

Comment:
Not sure the if/else is needed here, Nautobot models generally always have a PK value even before they're saved to the database. If there is a reason it's needed, `self.present_in_database` is a better check.